src/Symfony/Component/Form/Form.php

```diff
@@ -562,6 +562,7 @@ public function submit($submittedData, $clearMissing = true)
 
                 foreach ($this->children as $name => $child) {
                     $isSubmitted = array_key_exists($name, $submittedData);
+                    $clearMissing = $this->getConfig()->getOption('expanded') ?: $clearMissing;
```

Comment:
`expanded` being an option for `ChoiceType`, it looks weird to "hardcode" this condition here.